math/fibonacci_fast.cpp

```diff
@@ -19,28 +19,31 @@
 #include <cstdio>
 #include <iostream>
 
-/** maximum number that can be computed - The result after 93 cannot be stored
- * in a `uint64_t` data type. */
-const uint64_t MAX = 93;
+/**
+ * maximum number that can be computed - The result after 93 cannot be stored
+ * in a `uint64_t` data type.
+ */
 
-/** Array of computed fibonacci numbers */
-uint64_t f[MAX] = {0};
+#define MAX 93
 
 /** Algorithm */
 uint64_t fib(uint64_t n) {
-    if (n == 0)
+    static uint64_t f1 = 1,
+                    f2 = 1;  // using static keyword will retain the values of
+                             // f1 and f2 for the next function call.
+
+    if (n <= 2)
+        return f2;
+    if (n >= 93) {
+        std::err << "Cannot compute for n>93 due to limit of 64-bit integers\n";
```

Comment:
```suggestion
        std::cerr << "Cannot compute for n>93 due to limit of 64-bit integers\n";
```
otherwise, there would be an error